### ISSUE TITLE
Add a Fuel project template

### ISSUE THREAD

**Issue by ra0x3** [FuelLabs/sway#2280]
**First off, I'm not sure if this functionality already exists. I'm creating this issue to figure out exactly what needs to be done**

- [Per this comment](https://fuellabs.slack.com/archives/C02RU9MNVKN/p1657288110139299) it seems like we need a new template in the `sway/examples` directory that users can pull from to create a default Fuel project?
  - I'm saying "Fuel project" (but it might just be a Sway project?) because this project template would contain components of the Fuel ecosystem: Sway project/contract, indexer project, frontend project, tests, etc
- I'm thinking users would pull down the template using `forc template [name] --template-name [template_name]`
  - We would have to decide a name for the template
    - I don't feel strongly about a name (preferably something short, that indicates that this template includes components of the Fuel ecosystem)

Additional context:
- Update the fuel-indexer docs accordingly, with this new `forc template` command
- The project structure would resemble:

```
➜  my-project tree .
.
├── Forc.toml
├── contracts
│   └── main.sw
├── frontend
│   └── index.html
├── indexer
│   ├── contracts
│   │   └── my-project-abi.json
│   ├── lib.rs
│   ├── main.rs
│   ├── manifest.yaml
│   └── schema
│       └── my-project-schema.graphql
└── tests
    └── harness.rs
```
- I would also note that currently, this structure requires the user to copy the ABI JSON over into indexer/contracts
  - Would be better if they didn't have to copy

**Comment by ra0x3** [comment]
CC @mitchmindtree @JoshuaBatty

**Comment by adlerjohn** [comment]
I'm not sure this should be an "example," or that we want to put it in this repo. It's a kind of cross-repo project.

**Comment by ra0x3** [comment]
Yea @adlerjohn part of the issue here is I'm not sure exactly _where_ this would even go

**Comment by mitchmindtree** [comment]
While I can see a template along these lines being useful down the road, we may want to consider addressing https://github.com/FuelLabs/fuel-indexer/issues/89 before starting something like this.

We might add unnecessary churn by hosting a template like this too early, particularly while we're still in the process of working out how we want the fuel-indexer configuration to look and how we want to streamline the indexer devex more generally.

Perhaps we can continue to focus on improving the Indexer experience in the meantime and revisit this at a later point?

---

It's not clear to me that we want to do this via `forc template`. This project layout heavily implies a Rust-hosted application, so we may want to instead consider providing [a `cargo generate` template](https://github.com/cargo-generate/cargo-generate), at least to begin?

Cargo generate is quite a bit more powerful than `forc template` at the moment, has lots of Rust/cargo-aware features along with support for [the liquid template language](https://github.com/Shopify/liquid), support for hook scripts and regex placeholders. It also provides a little CLI for users to fill in certain details interactively as they initialize their project from the template.

While it would be nice to have more extensive features like this as a part of `forc template` in the future, cargo generate already solves these problems for Rust projects quite nicely.

---

> part of the issue here is I'm not sure exactly where this would even go

Yeah perhaps something like this might be best in its own dedicated repo as it has quite a few large cross-repo dependencies, (indexer, forc, something to build frontend, etc).

Maybe we could treat this like a downstream Fuel project skeleton demo or "reference repo" that brings together the different parts of the Fuel ecosystem (fuel-core, indexer, forc, frontend) under one example, and just happens to allow for it to be used as a template as well?
